manifests/plugin/dpdk_telemetry.pp

```diff
@@ -0,0 +1,28 @@
+# Class to manage dpdk_telemetry plugin for collectd.
+#
+# The dpdk_telemetry plugin collects DPDK ethernet device metrics via
+# dpdk_telemetry library.
+#
+# Plugin retrieves metrics from a DPDK packet forwarding application
+# by sending the JSON formatted message via a UNIX domain socket.
+# DPDK telemetry component will respond with a JSON formatted reply
+# delivering the requested metrics. Plugin parses the JSON data
+# and publishes the metric values to collectd for further use.
+#
+# @param ensure Ensure param for collectd::plugin type.
+# @param client_socket_path UNIX domain client socket to receive messages from DPDK telemetry library.
+# @param dpdk_socket_path UNIX domain DPDK telemetry socket to be connected to send messages.
+#
+class collectd::plugin::dpdk_telemetry (
+  Enum['present', 'absent'] $ensure             = 'present',
+  Optional[String]          $client_socket_path = '/var/run/.client',
```

Comment:
Use `Optional[Stdlib::Absolutepath]` instead as this value is either undef or a path to a file.

Reply:
if it's an `Optional[]` datatype, the default value has to be `undef`. If you want to pass a default path, like `/var/run/.client`, it doesn't need to be `Optional[]` because you cannot pass `undef` to it. Same applies for `$dpdk_socket_path`

Reply:
The default value could be, but does not have to be `undef`. Since the value is either `undef` or a path to a file, then `Stdlib::Absolutepath` makes more sense than `String`.

Reply:
I changed String to Stdlib:Absolutepath and kept "Optional[...]" Is that all right?

Reply:
@ghoneycutt if you pass undef to a variable that has a default value, undef won't overwrite it. In my opinion the `Optional[]` needs to be removed.

Reply:
I removed `Optional[]` since, as @bastelfreak  mentioned, we can't change it back to `undef` if we already have defined default value. Without `Optional[]` there is now no need for `*_socket_path => undef` test case.

Reply:
@ghoneycutt  are you satisfied with applied modifications?